mantle/cosa/build.go

```diff
@@ -38,7 +38,7 @@ func buildParser(r io.Reader) (*Build, error) {
 		return nil, errors.Wrapf(err, "failed to parse build")
 	}
 	if errs := cosaBuild.Validate(); len(errs) > 0 {
-		return nil, ErrMetaFailsValidation
+		return nil, errors.Wrapf(ErrMetaFailsValidation, "%v", errs)
```

Comment:
Minor: maybe `strings.Join(errs, ", ")` here? Makes it easier to tell apart the separate errors.

Reply:
Kind of, unless the error strings have a `,` in them.  Seems simpler as is, but I typed up what you wanted:

```
diff --git a/mantle/cosa/build.go b/mantle/cosa/build.go
index 362f7090..a27ca8c7 100644
--- a/mantle/cosa/build.go
+++ b/mantle/cosa/build.go
@@ -15,6 +15,7 @@
 package cosa
 
 import (
+	"strings"
 	"encoding/json"
 	"fmt"
 	"io"
@@ -38,7 +39,11 @@ func buildParser(r io.Reader) (*Build, error) {
 		return nil, errors.Wrapf(err, "failed to parse build")
 	}
 	if errs := cosaBuild.Validate(); len(errs) > 0 {
-		return nil, errors.Wrapf(ErrMetaFailsValidation, "%v", errs)
+		errstrs := []string{}
+		for _, err := range errs {
+			errstrs = append(errstrs, err.Error())
+		}
+		return nil, errors.Wrapf(ErrMetaFailsValidation, strings.Join(errstrs, ", "))
 	}
 	return cosaBuild, nil
 }
```

Dunno.